### PR TITLE
Adds name validation in the Geopolygon view and translation fix

### PR DESCRIPTION
**Pull request by ajeety4**
## Product Description

Adds name validation in the Geopolygon view and a translation fix in the JS file.
No User facing changes.

## Technical Summary

[Ticket](https://dimagi.atlassian.net/browse/SC-3785)

## Feature Flag

[Geospatial](https://staging.commcarehq.org/hq/flags/edit/geospatial/)

### Safety story

Small change, Local testing done

### Automated test coverage

Added new test case for the name validation

### QA Plan

None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/geospatial/static/geospatial/js/models.js [hunk 1]
```diff
@@ -692,7 +695,7 @@ hqDefine('geospatial/js/models', [
                     }, 2000);
                 },
                 error: function () {
-                    alertUser.alert_user(gettext(unexpectedErrorMessage), 'danger');
```
Comment (ajeety4) [inline corehq/apps/geospatial/static/geospatial/js/models.js:695]: Hey @mkangia ,
I was wrong about this when I mentioned this `gettext(unexpectedErrorMessage)` works.  The alert message is shown correctly however the text is not picked up for translation. 
It is now fixed in this commit.

File: corehq/apps/geospatial/views.py [hunk 2]
```diff
@@ -107,6 +107,9 @@ def post(self, request, *args, **kwargs):
             )
 
         geo_polygon_name = geo_json.pop('name')
+        if geo_polygon_name == '':
+            return HttpResponseBadRequest(_('Please specify name for the GeoPolygon area.'))
```
Comment (zandre-eng) [inline corehq/apps/geospatial/views.py]: Nit: Slight grammar change

```suggestion
            return HttpResponseBadRequest(_('Please specify a name for the GeoPolygon area.'))
```

**Review by zandre-eng — APPROVED** [review]
Looks good! Just one minor nit.

File: corehq/apps/geospatial/views.py (hunk 2 above)
Comment (ajeety4) [reply corehq/apps/geospatial/views.py]: Addressed in https://github.com/dimagi/commcare-hq/pull/35234/commits/2eb87294008b289d824c845cd06049dd42584bd8

**Review by zandre-eng — APPROVED** [review]

**Review by kaapstorm — APPROVED** [review]

**Pull request merged**